### PR TITLE
feat: deploy and manage a local instance

### PR DESCRIPTION
**Pull request by TobiaszCudnik**
Because

- user should be able to spin up a local instance

This commit

- adds deployment, management and auth (INS-1659) for a single local instance

```
$ instill local --help
Create and manage a local Instill Core instance with ease.

USAGE
  instill local <command> [flags]

CORE COMMANDS
  deploy:     Deploy a local Instill Core instance
  start:      Start a local Instill Core instance
  status:     Status of the local deployment
  stop:       Stop a local Instill Core instance

INHERITED FLAGS
  --help   Show help for command

LEARN MORE
  Use 'instill <command> <subcommand> --help' for more information about a command.
  Read the manual at https://docs.instill.tech
```

### CODE REVIEW

File: pkg/cmd/local/deploy.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		return fmt.Errorf("ERROR: wrong input, %w", err)
+	}
+
+	// check the deps
+	apps := []string{"docker", "make", "git"}
+	for _, n := range apps {
+		if opts.Exec != nil {
+			_, err = opts.Exec.LookPath(n)
+		} else {
+			_, err = opts.Exec.LookPath(n)
```
Comment (pinglin) [inline pkg/cmd/local/deploy.go:94]: This line causes
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x180646a]
```

**Review by pinglin — CHANGES_REQUESTED** [review]
Fatal error causing the local deployment failed. I will try to solve this on my side.

File: pkg/cmd/local/deploy.go (hunk 1 above)
Comment (TobiaszCudnik) [reply pkg/cmd/local/deploy.go]: Please include a full stack trace when reporting issues.

File: pkg/cmd/local/deploy.go [hunk 2]
```diff
@@ -64,11 +65,11 @@ func NewDeployCmd(f *cmdutil.Factory, runF func(*DeployOptions) error) *cobra.Co
...
 		},
 	}
-	pwd, err := os.Getwd()
+	d, err := os.UserHomeDir()
 	if err != nil {
-		logger.Error("Couldn't get pwd", err)
+		logger.Error("Couldn't get Home directory", err)
 	}
-	dir := filepath.Join(pwd, "instill-core") + string(os.PathSeparator)
+	dir := filepath.Join(d, ".config", "instill") + string(os.PathSeparator)
```
Comment (TobiaszCudnik) [inline pkg/cmd/local/deploy.go:72]: This is an incorrect location, it should be in `.local` or `.cache`.

**Pull request merged**